src/System.IO/tests/Stream/NullTests.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace StreamTests
+{
+    public class NullCtorTests
+    {
+        [Fact]
+        public static async Task TestNullStream()
+        {
+            Stream s = Stream.Null;
+
+            s.Flush();
+
+            Assert.Equal(-1, s.ReadByte());
+
+            s.WriteByte(5);
+            int n = s.Read(new byte[2], 0, 2);
+            Assert.Equal(0, n);
+            s.Write(new byte[2], 0, 2);
+
+            Assert.Equal(0, await s.ReadAsync(new byte[2], 0, 2));
+
+            s.Flush();
+            s.Dispose();
+        }
+
+        [Theory]
+        [MemberData("NullReaders")]
+        public static void TestNullTextReader(TextReader input)
+        {
+            StreamReader sr = input as StreamReader;
+
+            if (sr != null)
```

Comment:
Some but not all of the TextReader instances passed in are or derive from StreamReader, e.g. TextReader.Null is an instance of NullTextReader, which does not derive from StreamReader.  In contrast, StreamReader.Null returns a NullStreamReader which does derive from StreamReader.